lib/rubocop/cop/mixin/shareable_constant_value.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for dealing with Shareable Constant Value
+    module ShareableConstantValue
+      module_function
+
+      def recent_shareable_value?(node)
```

Comment:
This methods has bad performance (`O(n)`) so overall this cop could be `O(n^2)`.
Ideally fix the performance of this method, or else make sure that it is called *last*, i.e. only when it really looks like there's a `.freeze` missing.

Reply:
For now called this as last line to make the best effort on performance.

I believe the only way to make this method performant is to memoize, rather store the last instance shareable_constant_value was provided. Is there any place in the codebase where such memoization is already being done? I could refer that raise a future PR to improve the performance of this method

Reply:
You can use normal memoization (`@cache ||= ...`). To be clean, you can clear the cache in `def on_new_investigation`, but it's not necessary unless your cop `support_multiple_source?`.